### PR TITLE
koord-scheduler: make Coscheduling/ElasticQuota/DeviceShare plugins more flexible

### PR DESCRIPTION
**Pull request by xulinfei1996**
Signed-off-by: xulinfei.xlf <xulinfei.xlf@alibaba-inc.com>

### Ⅰ. Describe what this PR does
1、make some functions from func xxx() to var xxx = func()
2、deviceResourceNames -> DeviceResourceNames
3、nvidiaGpuExist -> NvidiaGpuExist and etc.

### V. Checklist

- [ ] I have written necessary docs and comments
- [ ] I have added necessary unit tests and integration tests
- [ ] All checks passed in `make test`

### CODE REVIEW

File: pkg/scheduler/plugins/deviceshare/utils.go [hunk 1]
```diff
@@ -146,31 +146,31 @@ func convertCommonDeviceResource(podRequest corev1.ResourceList, deviceType sche
 	return resources
 }
 
-// nvidia.com/gpu means applying for full-card
+// ConvertGPUResource nvidia.com/gpu means applying for full-card
 // koordinator.sh/gpu means applying for cards in percentile
-// convertGPUResource will convert either nvidia.com/gpu or koordinator.sh/gpu to koordinator.sh/gpu-core and koordinator.sh/gpu-memory-ratio
-func convertGPUResource(podRequest corev1.ResourceList, combination uint) corev1.ResourceList {
+// ConvertGPUResource will convert either nvidia.com/gpu or koordinator.sh/gpu to koordinator.sh/gpu-core and koordinator.sh/gpu-memory-ratio
+var ConvertGPUResource = func(podRequest corev1.ResourceList, combination uint) corev1.ResourceList {
```
Comment (jasonliu747) [inline pkg/scheduler/plugins/deviceshare/utils.go:152]: `func ConvertGPUResource`?

File: pkg/scheduler/plugins/deviceshare/utils.go [hunk 2]
```diff
@@ -146,31 +146,31 @@ func convertCommonDeviceResource(podRequest corev1.ResourceList, deviceType sche
 	return resources
 }
 
-// nvidia.com/gpu means applying for full-card
+// ConvertGPUResource nvidia.com/gpu means applying for full-card
```
Comment (jasonliu747) [inline pkg/scheduler/plugins/deviceshare/utils.go:149]: 这一行注释不需要改

File: apis/extension/elastic_quota.go [hunk 3]
```diff
@@ -76,3 +76,7 @@ func IsForbiddenModify(quota *v1alpha1.ElasticQuota) (bool, error) {
 
 	return false, nil
 }
+
+var GetQuotaName = func(pod *corev1.Pod) string {
```
Comment (jasonliu747) [inline apis/extension/elastic_quota.go:80]: func GetQuotaName

File: pkg/scheduler/plugins/deviceshare/utils.go (hunk 2 above)
Comment (xulinfei1996) [reply pkg/scheduler/plugins/deviceshare/utils.go]: fixed

**Review by eahydra — APPROVED** [review]
/lgtm

**Review by yihuifeng — COMMENTED** [review]
/lgtm

**Review by buptcozy — APPROVED** [review]
/lgtm

**Pull request merged**